roles/matrix-synapse/templates/synapse/systemd/matrix-synapse.service.j2

```diff
@@ -43,6 +43,18 @@ ExecStart={{ matrix_host_command_docker }} run --rm --name matrix-synapse \
 			{% if matrix_synapse_manhole_enabled and matrix_synapse_container_manhole_api_host_bind_port %}
 			-p {{ matrix_synapse_container_manhole_api_host_bind_port }}:9000 \
 			{% endif %}
+			{% for worker in matrix_synapse_workers_enabled_list %}
+			{% if matrix_synapse_workers_enabled and not matrix_nginx_proxy_enabled|default(False) %}
+			{# Expose worker ports (by default 18xxx range) on host if not using internal nginx proxy #}
```

Comment:
I am not sure what exactly this does, why would worker ports be mapped in the main synapse container (back to the host)? Shouldn't the mapping be (workers|synapse)->host (and therefore the worker mapping be seperate from synapse port mappings)?

Reply:
Because the worker processes run in the synapse container alongside the main homeserver process. When not using the builtin nginx container instance, they have to be accessible by a process outside the matrix docker network, thus exposing the ports on the host.

If you think it would improve things if worker ran in its own container, please test that.